### DISCUSSION TITLE
How to fix TaskbarNotification in MaterialDesignInXamlToolkit?

### DISCUSSION THREAD

**Discussion by VahidEra** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3101] [Q&A] [answered]
Hi guys,
I am testing notifyIcon in wpf based on MaterialDesignInXamlToolkit and in this I am using harcodet wpf taskbar notification dll ,because material design does not apply to create notifiyIcon through form.NotifyIcon .  But there is a problem. Unfortunately, after applying these changes, when I specify a name for the notification, its tooltip is displayed at the top of the right of the screen.  Is this a material design bug and nothing can be done or do you have a suggestion for this?  Thanks for your help.

![2023-02-24_171833](https://user-images.githubusercontent.com/27216710/221237705-1b1be2c6-158b-40db-adca-535bc479793e.png)
 

I have uploaded the project file for your convenience and better understanding:
https://drive.google.com/file/d/1eUttXAL1lBiJGkekssUEmVXeAPQrAYdg/view?usp=sharing

**Comment by VahidEra** [comment, marked as answer]
@miloush Dear Jan,My dear friend What do you think about this? Based on your knowledge and skills, I felt that I would like to ask you for your opinion.

**Reply by miloush** [reply]
As a general tip when reporting issues, include instructions how people can reproduce what you are seeing and what you expect to see instead. Furthermore, it would help to try to convince anyone from the repo you are filing the issue at that it is their issue rather than anyone else's. A good start is checking and saying that if you remove the material design theme, the issue does not happen.

The reason for the issue is that the `MaterialDesignToolTip` overrides the default tooltip placement from `Mouse` to `Custom`:

https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/8c705167db77b7833fa19acb42dc3961850cb82b/MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ToolTip.xaml#L9

and has its own logic how to place tooltips. To be fair it does not seem that WPF's custom placement callback lets consumers easily detect that there is no placement target to deal with this situation, which is something that could be brought up in the WPF repo.

Anyway, to workaround it, you will need to revert the tooltip placement back to `Mouse`. You can do that, again, by explicitly declaring the tooltip so that you can set its properties (because locally set properties have precedence over values from styles):

```xml
<tb:TaskbarIcon.TrayToolTip>
    <ToolTip Content="Fallback ToolTip for Windows xp" Placement="Mouse" />
</tb:TaskbarIcon.TrayToolTip>
```

**Reply by VahidEra** [reply]
@miloush, Dear Jan You are completely right.  Thank you for your advice, I will definitely consider these things.
Again, as always, you solved my problem, I appreciate your honest help.
